Source/WebCore/style/values/grid/StyleGridTemplateAreas.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "StyleGridTemplateAreas.h"
+
+#include "CSSGridTemplateAreasValue.h"
+#include "GridPositionsResolver.h"
+#include "StyleBuilderChecking.h"
+#include "StylePrimitiveNumericTypes+Blending.h"
+#include <algorithm>
+
+namespace WebCore {
+namespace Style {
+
+static GridNamedLinesMap initializeImplicitNamedGridLines(const GridNamedAreaMap& namedGridAreas, GridTrackSizingDirection direction)
+{
+    GridNamedLinesMap namedGridLines;
+
+    for (auto& area : namedGridAreas.map) {
+        auto areaSpan = direction == GridTrackSizingDirection::ForRows ? area.value.rows : area.value.columns;
+        {
+            auto& startVector = namedGridLines.map.add(makeString(area.key, "-start"_s), Vector<unsigned>()).iterator->value;
+            startVector.append(areaSpan.startLine());
+            std::ranges::sort(startVector);
```

Comment:
Adding a number and keeping an already-sorted vector sorted is not most efficiently done with `append` and `sort`. We should use the better algorithm instead, more like this, although this is not entirely correct:
```
template<typename ContainerType, typedef ValueType>
typename void insertSorted(ContainerType& container, ValueType&& value)
{
    ASSERT(std::ranges::is_sorted(container));
    auto location = std::ranges::upper_bound(container, value);
    return container.insert(location, std::forward<ValueType>(value));
}
```

Reply:
Going to leave this as is for this change. Really, I think we can just sort once at the end here, but I want to keep that kind of thing separate.